lambda_functions/two_way_sms/two_way_sms_v2.py

```diff
@@ -0,0 +1,428 @@
+            for record in event_data.get('Records'):
+                if record.get('body') is None:
+                    # Log specific record causing issues
+                    logger.critical('Failed to retrieve body in record: %s', record)
+                    return False
+            return True
+    except Exception as e:
+        logger.critical('Failed to parse event_data %s', e)
+        return False
+
+    return False        
+
+def forward_to_service(inbound_sms: dict, url: str) -> bool:
+    """
+    Forwards inbound SMS to the service that has 2-way SMS setup.
+    """
+    try:
+        if url is None:
+            logger.error("No URL provided in configuration for service")
+            return False
+
+        logger.debug('Connecting to %s, sending: %s', url, inbound_sms)
+
+        headers = {
+            'Content-type': 'application/json'
+        }
+
+        response = requests.post(
+            url,            
+            verify=False,
+            json=inbound_sms,
+            timeout=TIMEOUT,
+            headers=headers
+        )
+
+        logger.info('Response Status: %d', response.status_code)
+        logger.debug('Response Content: %s', response.content)
+
+        return True
+    except requests.HTTPError as e:
+        logger.error("HTTPError With Http Request")
+        logger.exception('%s', e)
+    except requests.RequestException as e:
+        logger.error("RequestException With Http Request")
+        logger.exception('%s', e)
+    except Exception as e:
+        logger.error("General Exception With Http Request")
+        logger.exception('%s', e)
+        # Need to raise here to move the message to the dead letter queue instead of retry. 
+        raise
+
+    return False
+
+def push_to_sqs(inbound_sms: dict, is_retry: bool) -> None:
+    """
+    Pushes an inbound sms or entire event to SQS. Sends to RETRY or DEAD LETTER queue dependent
+    on is_retry variable. 
+    """
+    # NOTE: https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SendMessage.html
+    # Need to ensure none of those unicode characters are in the message or it's gone
+    try:
+        logger.warning('Pushing event to %s queue', "RETRY" if is_retry else "DEAD LETTER")
+        logger.debug('Event: %s', inbound_sms)
+
+        queue_msg = json.dumps(inbound_sms)
+
+        aws_sqs_client.send_message(QueueUrl=RETRY_SQS_URL if is_retry else DEAD_LETTER_SQS_URL,
+                                    MessageBody=queue_msg)
+
+        logger.info('Completed enqueue of message')
+    except Exception as e:
+        logger.exception('%s', e)
+        logger.critical('Failed to push event to SQS: %s', inbound_sms)
+        if is_retry:
+            # Push to dead letter queue if push to retry fails
+            push_to_sqs(inbound_sms, False)
+        else:
+            logger.critical('Attempt to enqueue to DEAD LETTER failed')
+
+# **Note** - Commented out because it wont be necessary in this initial release
```

Comment:
Maybe we could add a TODO to this big section of commented out code and link it to a future ticket so we know what it's going to be used for? Does the ticket to implement this code already exist?

Reply:
No it doesnt.  I like your idea though.

Reply:
https://github.com/department-of-veterans-affairs/notification-api/issues/1008